src/explore-education-statistics-common/src/modules/table-tool/components/FiltersForm.tsx

```diff
@@ -105,10 +115,33 @@ const FiltersForm = (props: Props & InjectedWizardProps) => {
     </WizardStepHeading>
   );
 
-  const handleSubmit = useFormSubmit(async (values: FormValues) => {
-    await onSubmit(values);
-    goToNextStep();
-  }, errorMappings);
+  const handleSubmit = async (values: FormValues) => {
+    setPreviousValues(values);
+    try {
+      await onSubmit(values);
+      toggleTableSizeError.off();
+      goToNextStep();
+    } catch (error) {
+      if (isServerValidationError(error) && error.response?.data) {
+        const errorMessages = Object.entries(error.response.data.errors).reduce(
+          (acc, [, messages]) => {
+            messages.forEach(message => acc.push(message));
+            return acc;
+          },
+          [] as string[],
+        );
+        if (errorMessages.includes('QUERY_EXCEEDS_MAX_ALLOWABLE_TABLE_SIZE')) {
```

Comment:
We seem to be doing a lot of work here when we it can probably boil it down to something like this:

```ts
if (isServerValidationError(error)) {
  const errors = Object.values(error.response?.data.errors);
  
  if (errors.includes('QUERY_EXCEEDS_MAX_ALLOWABLE_TABLE_SIZE')) {
    // do stuff... 
```

Reply:
Happy that this has been addressed :)